agent/consul/catalog_endpoint.go

```diff
@@ -20,74 +20,98 @@ type Catalog struct {
 	srv *Server
 }
 
-// Register is used register that a node is providing a given service.
-func (c *Catalog) Register(args *structs.RegisterRequest, reply *struct{}) error {
-	if done, err := c.srv.forward("Catalog.Register", args, args, reply); done {
+// nodePreApply does the verification of a node before it is applied to Raft.
+func nodePreApply(nodeName, nodeID string) error {
```

Comment:
I didn't do this yet to avoid copying fields from RegisterRequest into a Node struct in Catalog.Register. That way if this function needs more args in the future it'll be obvious where to add them (where otherwise there would be at least one place in the code passing in a Node with potentially empty fields that would fail silently).